src/module-path.ts

```diff
@@ -24,7 +24,7 @@ export function getModulePath(currentModule: string, importPath: string, host: I
     const pkgFile = host.joinPath(dir, 'package.json');
     if (host.fileExists(pkgFile) && host.isFile(pkgFile)) {
       const pkg = JSON.parse(host.readFile(pkgFile).toString());
-      const main = pkg['jsnext:main'] || (typeof pkg.browser === 'string' ? pkg.browser : undefined) || pkg.main;
+      const main = (typeof pkg.browser === 'string' ? pkg.browser : undefined) || pkg['jsnext:main'] || pkg.main;
```

Comment:
* I'd consider changing the behavior to `key in pkg ? use : use-other` as falsy values have distinct meaning for `browser` and `jsnext:main` fields 
* `browser` can be an object, too:
https://github.com/defunctzombie/package-browser-field-spec#replace-specific-files---advanced.